internal/pkg/create/create.go

```diff
@@ -225,7 +226,8 @@ func createApp(ctx context.Context, dirPath string, template Template, gitBranch
 	if template.isGit {
 		doctorSection, err := doctor.CheckGit(ctx)
 		if doctorSection.HasError() || err != nil {
-			zipFileURL := generateGitZipFileURL(template.path, gitBranch)
+			httpClient := slackdeps.NewHTTPClient(slackdeps.HTTPClientOptions{})
+			zipFileURL := generateGitZipFileURL(httpClient, template.path, gitBranch)
```

Comment:
note: This is the main code change. We now inject a `httpClient` and mock it in tests.

We could take this further by defining a `clients.HTTPClient` instance, but I wanted to keep these changes scoped smaller to allow us to discuss those trade-offs later.